src/utils.cc

```diff
@@ -740,14 +757,31 @@ bool utils::anyParameterAtBoundaries( const RooArgSet &params, int verbosity ){
 
     RooLinkedListIter iter = params.iterator(); int i = 0;
     for (RooRealVar *a = (RooRealVar *) iter.Next(); a != 0; a = (RooRealVar *) iter.Next(), ++i) {
-
-        bool isBad = isParameterAtBoundary(*a);
+      
+        bool atLowerBoundary = isParameterAtLowerBoundary(*a);
+        bool atUpperBoundary = isParameterAtLowerBoundary(*a);
+        bool isBad = atLowerBoundary || atUpperBoundary;
 
         if(isBad){
             std::string varName((*a).GetName());
 
             if( verbosity >= 9 || (timesFoundAtBoundary[varName] < 3 && verbosity > -1) ){
-                fprintf(CloseCoutSentry::trueStdOutGlobal(),"  [WARNING] Found [%s] at boundary. \n", (*a).GetName());
+              
+                fprintf(CloseCoutSentry::trueStdOutGlobal(),"  [WARNING] Found [%s] at ", (*a).GetName());
+
+                if (atLowerBoundary && atUpperBoundary)
+                  fprintf(CloseCoutSentry::trueStdOutGlobal(),"both boundaries");
+                else if (atLowerBoundary)
+                  fprintf(CloseCoutSentry::trueStdOutGlobal(),"lower boundary");
+                else 
+                  // must be at upper boundary
```

Comment:
On 06/07/15 04:10, André David wrote:
>
> In src/utils.cc 
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/197#discussion_r33902531>:
>
> >
> >          if(isBad){
> >              std::string varName((*a).GetName());
> >
> >              if( verbosity >= 9 || (timesFoundAtBoundary[varName] < 3 && verbosity > -1) ){
> > -                fprintf(CloseCoutSentry::trueStdOutGlobal(),"  [WARNING] Found [%s] at boundary. \n", (*a).GetName());
> > +
> > +                fprintf(CloseCoutSentry::trueStdOutGlobal(),"  [WARNING] Found [%s] at ", (*a).GetName());
> > +
> > +                if (atLowerBoundary && atUpperBoundary)
> > +                  fprintf(CloseCoutSentry::trueStdOutGlobal(),"both boundaries");
> > +                else if (atLowerBoundary)
> > +                  fprintf(CloseCoutSentry::trueStdOutGlobal(),"lower boundary");
> > +                else
> > +                  // must be at upper boundary
>
> It would be better to not trust, but rather test everything and if not 
> fitting any case, give some meaningful message.
>
I don't understand this comment; this is a simple extension of the 
original code. In fact,

bool isBad = atLowerBoundary || atUpperBoundary;

so when isBad is true and (atLowerBoundary && atUpperBoundary) is false 
and atLowerBoundary is false
the only possibility is that atUpperBoundary is true (otherwise isBad 
would be false).